dnf5/main.cpp

```diff
@@ -634,6 +642,7 @@ int main(int argc, char * argv[]) try {
     }
 
     auto command = context.get_selected_command();
+    bool any_repos_from_system_configuration = false;
```

Comment:
May I ask you for add a description what it means or what it contains. I think it should be `true` when any repository is created from .repo, configuration option - `--repofrompath`, or commandline packages were specified argument `<>.rpm`.

Reply:
Thanks, added a comment here.

If the user specifies a nonexistent RPM on the command line, this variable will not get set to true, and a warning message will still get printed:

```
$ dnf5 --installroot=/mnt install -y ./missing.rpm
No repositories were loaded from the installroot. To use the configuration and repositories of the host system, pass --use-host-environment.
Failed to resolve the transaction:
No match for argument: ./missing.rpm
```

but I think that's ok. It will still not print if the file is present and the transaction succeeds.

Reply:
In DNF4 - please take a look into logic of `def _checkEnabledRepo(base, possible_local_files=()):` in `dnf/cli/commands/__init__.py`. The logic there does not report problem with repos when rpms are defined by argument. Related bug - https://bugzilla.redhat.com/show_bug.cgi?id=620802. I an suggesting to add a TODO - implement check from DNF4 in `def _checkEnabledRepo()` and integrate this warning in the method. What do you think?

Reply:
I am sorry - please add a TODO not here but around line 731.
I have `any_repos_from_system_configuration` part `system_configuration` is confusing, because repository could be created from commandline `--repofrompath`, or by plugin.  What about name - `any_available_repository`

Reply:
Hmm, I just checked again and I now get the proper error from `dnf5 --installroot=/mnt install -y ./missing.rpm`: 
```
Updating and loading repositories:
Repositories loaded.
Failed to access RPM "./missing.rpm": No such file or directory
```
Maybe something changed when I rebased.

Anyway, I think we do actually want to show the warning even when RPMs are specified by an argument. Consider the following case:
```
~/git/dnf5 $ sudo ./build/dnf5/dnf5 --installroot=/tmp install -y hello-2.12.1-1.2.x86_64.rpm 
Updating and loading repositories:
Repositories loaded.
No repositories were loaded from the installroot. To use the configuration and repositories of the host system, pass --use-host-config.
Failed to resolve the transaction:
Problem: conflicting requests
  - nothing provides libc.so.6()(64bit) needed by hello-2.12.1-1.2.x86_64
  - nothing provides libc.so.6(GLIBC_2.2.5)(64bit) needed by hello-2.12.1-1.2.x86_64
  - nothing provides libc.so.6(GLIBC_2.3)(64bit) needed by hello-2.12.1-1.2.x86_64
  - nothing provides libc.so.6(GLIBC_2.3.4)(64bit) needed by hello-2.12.1-1.2.x86_64
  - nothing provides libc.so.6(GLIBC_2.34)(64bit) needed by hello-2.12.1-1.2.x86_64
  - nothing provides libc.so.6(GLIBC_2.4)(64bit) needed by hello-2.12.1-1.2.x86_64
```

I specify a valid RPM on the command line, but its dependencies can't be found because no other repositories are loaded. I think it is desirable in this case to show the warning, no?